### DISCUSSION TITLE
Client not switching back to synchronize

### DISCUSSION THREAD

**Discussion by FaehlingT** [abraunegg/onedrive#2469] [Q&A] [answered]
I initiated the client without problems in --monitor mode, but now that all my files have synced and I don't get 429 errors anymore, I would like to switch to --synchronize.
So I did the following commands:
```
systemctl --user stop onedrive; onedrive --synchronize --resync; systemctl --user start onedrive;
```
but it still prints out:
```
2204 /usr/local/bin/onedrive --monitor
```
What am I doing wrong?

**Comment by abraunegg** [comment, marked as answer]
As per https://github.com/abraunegg/onedrive/issues/2468 

The application output is clear - you still have an active instance running in the background - most likely via an extra systemd service as you are running ~Crapbuntu~ Ubuntu.

To potentially resolve, you should do the following:
1. Uninstall your manually compiled application version. For current ~Crapbuntu~ Ubuntu platforms you should be installing from the OpenSuSE Build Service. 
2. Uninstall any Ubuntu Universe 'onedrive' package you have installed.
3. Remove all systemd services that you have setup for this application. Note - ~Crapbuntu~ Universe Ubuntu packages have a bug where the package installs a second systemd service that you did not specify and/or start .... 
4. Install the 'onedrive' client the *correct* way for ~Crapbuntu~ Ubuntu as per the [installation guide](https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md): 

![image](https://github.com/abraunegg/onedrive/assets/4956234/69178865-c86f-4b54-a039-a2eedf97c318)

Following the [correct instructions](https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md) will get you the client correctly installed on your ~Crapbuntu~ Ubuntu system. Following each step will also clean out any erroneous systemd service file that may still exist.

Once this is done, you can configure the application to run as a system service by following [this advice](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#running-onedrive-as-a-system-service) as per the [usage document](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md) 

Once this is done, to correctly stop, perform a manual sync, then restart the client, this is required:
```
systemctl --user stop onedrive; onedrive --synchronize; systemctl --user start onedrive;
```

Please do not use `--resync` unless the application explicitly tells you to use it. There are still *so* many bad references out there on the Internet that say to use `--resync` ... do not use this option flag unless you change application configuration and the application tells you to use it. By using that option it will be contributing heavily to you getting 429 responses from the OneDrive API.

Now .. if this is *still* causing the original issue - what potentially is happening is that the systemd service is being slow to execute the service stop command and the application is taking time to process any database commits. Try each command separately and see what happens to potentially overcome any systemd processing issue.

The other aspect here is - you need to ask yourself why do you want to perform a manual sync? Your systemd service is performing a sync operation by default every 5 minutes and will pull down any new data it finds online. Any new files created locally will be uploaded automatically. 

If it is simply because you want to see or understand if things are in sync, there are other commands available to do this, and/or increase your application logging level using `--verbose` and have the application write all application output to a separate file that you can review and/or log rotate as needed.
